src/Microsoft.AspNet.OData.Shared/PatchMethodHandler.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.OData
+{    
+    /// <summary>
+    /// Basic Interface for PatchMethodHAndler
+    /// </summary>
+    public interface IPatchMethodHandler
+    {
+
+    }
+
+    /// <summary>
+    /// Handler Class to handle users methods for create, delete and update
+    /// </summary>
+    public abstract class PatchMethodHandler<TStructuralType>: IPatchMethodHandler
+    {
+        /// <summary>
+        /// TryCreate method to create a new object.
+        /// </summary>        
+        /// <param name="createdObject">The created object (CLR or Typeless)</param>
+        /// <param name="errorMessage">Any error message in case of an exception</param>
+        /// <returns></returns>
+        public abstract PatchStatus TryCreate(out TStructuralType createdObject, out string errorMessage);
```

Comment:
>TryCreate [](start = 36, length = 9)

Probably need to pass in the item being created -- it may be invalid to create an object with all null/default values.